src/dbnode/persist/fs/commitlog/commit_log.go

```diff
@@ -442,19 +469,24 @@ func (l *commitLog) write() {
 				}
 			}
 
-			if isRotateLogsEvent {
-				write.callbackFn(callbackResult{
-					eventType: write.eventType,
-					err:       err,
-					rotateLogs: rotateLogsResult{
-						file: file,
-					},
-				})
+			var file persist.CommitLogFile
+			if err == nil {
+				// The contract for the RotateLogs() API is that it will return the
+				// commitlog file that became the primary one as a direct result of
+				// the method call. The new active file corresponds to index zero and
+				// the new secondary file corresponds to index 1.
+				file = files[0]
```

Comment:
Should this be a `const` at the top and explained there instead of here?

Reply:
I don't know how I feel about making it a const....if anything I'd maybe want to change the openWriters() API to return two separate arguments or something

Reply:
Why don't you make `openWriters` instead just return both writers as args?

i.e.
```go
primaryFile, secondaryFile, err := l.openWriters()

// then you don't need the second one most of the time
// i.e.
primaryFile, _, err := l.openWriters()
```